sdk/ai/ai-inference-rest/test/public/utils/recordedClient.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  Recorder,
+  RecorderStartOptions,
+  VitestTestContext,
+  assertEnvironmentVariable,
+} from "@azure-tools/test-recorder";
+import { AzureKeyCredential } from "@azure/core-auth";
+import { ClientOptions } from "@azure-rest/core-client";
+import createClient, { ModelClient } from "../../../src/index.js";
+
+const envSetupForPlayback: Record<string, string> = {
+  AZURE_ENDPOINT: "https://endpoint",
+  AZURE_CLIENT_ID: "azure_client_id",
+  AZURE_CLIENT_SECRET: "azure_client_secret",
+  AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
+  SUBSCRIPTION_ID: "azure_subscription_id",
+};
+
+const recorderEnvSetup: RecorderStartOptions = {
+  envSetupForPlayback,
+};
+
+/**
+ * creates the recorder and reads the environment variables from the `.env` file.
+ * Should be called first in the test suite to make sure environment variables are
+ * read before they are being used.
+ */
+export async function createRecorder(context: VitestTestContext): Promise<Recorder> {
+  const recorder = new Recorder(context);
+  await recorder.start(recorderEnvSetup);
+  return recorder;
+}
+
+export async function createModelClient(
+  recorder?: Recorder,
+  options?: ClientOptions,
+): Promise<ModelClient> {
+  const endpoint = assertEnvironmentVariable("AZURE_ENDPOINT");
+  const apikey = assertEnvironmentVariable("AZURE_CLIENT_SECRET");
+  const credential = new AzureKeyCredential(apikey);
```

Comment:
could we use token credential?

Reply:
When written, I wasn't able to get Entra ID working. I'll try again and update if it works now.

Reply:
[Updated](https://github.com/Azure/azure-sdk-for-js/pull/29863/commits/f5e56850c96fedc73e117073d2e9c567ee9ee918)